File: android/app/src/main/kotlin/com/anhquan/unisync/core/Device.kt
package com.anhquan.unisync.core

import android.content.Context
import com.anhquan.unisync.models.DeviceInfo
import com.anhquan.unisync.models.DeviceMessage

class Device(val context: Context, val connection: DeviceConnection, val info: DeviceInfo) :
    DeviceConnection.ConnectionListener {
    init {
        connection.addMessageListener(this)
    }

    private val pairingHandler = PairingHandler(this)

    override fun onMessage(message: DeviceMessage) {
        pairingHandler.onMessageReceived(message)
        if (pairingHandler.state == PairingHandler.PairState.PAIRED) {

        }
    }
}

File: unisync_backend/lib/core/device.dart
import 'package:unisync_backend/core/device_connection.dart';
import 'package:unisync_backend/core/pairing_handler.dart';
import 'package:unisync_backend/models/device_info/device_info.model.dart';
import 'package:unisync_backend/models/device_message/device_message.model.dart';

class Device extends ConnectionListener {
  Device(this.connection) {
    connection.messageListener = this;
  }

  final DeviceConnection connection;
  late final DeviceInfo info = connection.info;

  late final PairingHandler _pairingHandler = PairingHandler(this);

  @override
  void onMessage(DeviceMessage message) {
    _pairingHandler.onMessageReceived(message);
    if (_pairingHandler.state == PairState.paired) {}
  }
}
